### PR TITLE
feat: update Windows base images to 2024.05B

### PR DESCRIPTION
**Pull request by ShiqianTao**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

- WS2019: [May 14, 2024—KB5037765 (OS Build 17763.5820)](https://support.microsoft.com/en-us/topic/may-14-2024-kb5037765-os-build-17763-5820-82d1aefb-093c-4e4a-a729-cd4a829750ad)
- WS2022: [May 14, 2024—KB5037782 (OS Build 20348.2461)](https://support.microsoft.com/en-us/topic/may-14-2024-kb5037782-os-build-20348-2461-e4109269-b248-49cc-84fb-c2d0e2c86b36)
- WS23H2: [May 14, 2024—KB5037781 (OS Build 25398.887)](https://support.microsoft.com/en-us/topic/may-14-2024-kb5037781-os-build-25398-887-c10d3156-5ea0-41b7-a754-8ae1463ecc78)

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the help.

**Review by andyzhangx — APPROVED** [review]
lgtm in azuredisk-csi image version removal

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 [hunk 1]
```diff
@@ -97,27 +95,21 @@ $global:imagesToPull += @(
...
     "mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.28.9", # for k8s 1.28.x
-    "mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.29.3", # for k8s 1.29.x
     "mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.29.4", # for k8s 1.29.x
+    "mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.29.5", # for k8s 1.29.x
     "mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.30.0", # for k8s 1.30.x
+    "mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.30.1", # for k8s 1.30.x
     # OMS-Agent (Azure monitor). Owner: ganga1980 (Ganga Mahesh Siddem)
     "mcr.microsoft.com/azuremonitor/containerinsights/ciprod:win-3.1.20",
-    # CNS (Container Networking Service) Owner: jaer-tsun (Jaeryn)
+    # CNS (Container Networking Service) Owner: jaer-tsun (Jaeryn) # aks-rp/toggles/global/azure-cni-cns-windows.yaml
```
Comment (AbelHu) [inline vhdbuilder/packer/generate-windows-vhd-configuration.ps1:111]: please delete the comment

**Review by AbelHu — CHANGES_REQUESTED** [review]
Added some comments

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 [hunk 2]
```diff
@@ -97,27 +95,21 @@ $global:imagesToPull += @(
...
     # Addon of Azure secrets store. Owner: jiashun0011 (Jiashun Liu)
-    "mcr.microsoft.com/oss/kubernetes-csi/secrets-store/driver:v1.3.4",
     "mcr.microsoft.com/oss/kubernetes-csi/secrets-store/driver:v1.4.2",
     "mcr.microsoft.com/oss/kubernetes-csi/secrets-store/driver:v1.4.3",
-    "mcr.microsoft.com/oss/azure/secrets-store/provider-azure:v1.4.1",
     "mcr.microsoft.com/oss/azure/secrets-store/provider-azure:v1.5.1",
     "mcr.microsoft.com/oss/azure/secrets-store/provider-azure:v1.5.2",
     # Azure cloud node manager. Owner: nilo19 (Qi Ni), lzhecheng (Zhecheng Li)
-    "mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.25.24", # for k8s 1.25.x
-    "mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.26.22", # for k8s 1.26.x
```
Comment (AbelHu) [inline vhdbuilder/packer/generate-windows-vhd-configuration.ps1:108]: keep 1.26

Comment (ShiqianTao) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:108]: recovered it, thank you

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 [hunk 3]
```diff
@@ -40,17 +40,17 @@ $global:defenderUpdateInfoUrl = "https://go.microsoft.com/fwlink/?linkid=870379&
...
             "mcr.microsoft.com/windows/nanoserver:1809"
         )
     }
     "2022-containerd*" {
-        $global:patchUrls = @("https://catalog.s.download.windowsupdate.com/c/msdownload/update/software/secu/2024/04/windows10.0-kb5036909-x64_786040b0b0d000b17d6a727ea93ff77d733d1044.msu")
-        $global:patchIDs = @("KB5036909")
+        $global:patchUrls = @("https://catalog.s.download.windowsupdate.com/c/msdownload/update/software/secu/2024/05/windows10.0-kb5037782-x64_a28aa2576fc6b120b127acfbb901d3546ba9db82.msu")
+        $global:patchIDs = @("KB5037782")
 
         $global:imagesToPull = @(
```
[the remark this answers is not in the document]
Comment (ShiqianTao) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1:55]: yes, 23h2 has pre-installed the msu

File: vhdbuilder/packer/generate-windows-vhd-configuration.ps1 (hunk 1 above)
Comment (ShiqianTao) [reply vhdbuilder/packer/generate-windows-vhd-configuration.ps1]: deleted, thank you.

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR

**Pull request merged**